text/3730-nit-lint-level.md

```diff
@@ -0,0 +1,378 @@
+- Weak Warning: Code fragments that can be improved or optimized (redundant code, duplicated code fragments, and so on)
+- Server Problem
+- Grammar Error
+- Typo
+- Consideration: Code fragments that can be improved. This severity level is not marked on the error stripe and does not have a default highlighting style, but you can select one from the list of existing styles or configure your own.
+- No highlighting (but fix available)
+
+LSP has the following [diagnostic levels](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#diagnostic):
+- Error
+- Warning
+- Information
+- Hint
+
+with the following diagnostic tags:
+- Unnecessary
+- Deprecated
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- `CARGO_BUILD_NITS=nit` is modeled off of `CARGO_BUILD_WARNINGS=warn` which is just saying "treat X level violations as Y level"
+  - `CARGO_BUILD_WARNINGS=warn` is "weird" but shouldn't show up too often as its the default
+  - However, the frequency is swapped with `CARGO_BUILD_NITS=nit`
+  - We may also want the terms to be consistent across the different level-control fields
+  - Reminder: `CARGO_BUILD_WARNINGS` is unstable and we can still change it
+
+## Naming
+
+The name should clearly communicate that there is no authoriative weight
+encouraging them to resolve them but that it is left to them and their reviewer
+to weigh whether to resolve or ignore them.
+
+The verb form is used for:
+- attribute
+- long flag
+
+Lint names are expected to be read with their level for what should be done, e.g.
+- `deny(let_and_return)` -> "deny the use of let-and-return in this code"
+- `warn(let_and_return)` -> "warn on the use of let-and-return in this code"
+
+The noun form is used for:
+- Dynamic lint group
+
+Options
+- `#[nit]` / `--nit` / `-N` / `-Wnits`
+  - Unsure how to word `nit(let_and_return)` as a sentence like the other lint levels
+- `#[note]` / `--note` / `-N` / `-Wnotes`
+  - "note the let-and-return in this code"
+- `#[notice]` / `--notice` / `-N` / `-Wnotices`
+  - "notice the let-and-return in this code"
+- `#[mention]` / `--mention` / `-M` / `-Wmentions`
+  - "mention the let-and-return in this code"
+- `#[consider]` / `--consider` / ~~`-C` (taken)~~ / `-Wconsiderations` (IntelliJ)
+  - "consider the let-and-return in this code"
+  - Verb and noun have a larger divergence
+- `#[inform]` / `--inform` / `-I` / `-Winformation` (LSP)
+  - "inform you of the let-and-return in this code"
+  - Verb and noun have a larger divergence
+- `#[hint]` / `--hint` / `-H` / `-Whints` (LSP)
+  - "hint of a let-and-return in this code"
+- `#[remark]` / `--remark` / `-R` (supposedly LLVM) / `-Wremarks`
+  - "remark on the let-and-return in this code"
+- ~~`#[suggest]` / `--suggest` / `-S` / `-Wsuggestions`~~
+  - When read with the lint name, it sounds like its to find where you *should* do it rather than finding where you are doing it
+  - Verb and noun have a larger divergence
+
```

Comment:
last option I can think of: `cue`, which for me, it's quite an empty word, not used much, and it's not that far from hint/suggest.

> cue on the let-and-return in this code